packages/e2e-tests/specs/performance/site-editor.test.js

```diff
@@ -90,6 +90,7 @@ describe( 'Site Editor Performance', () => {
 		await visitSiteEditor( {
 			postId: id,
 			postType: 'page',
+			path: '/navigation/single',
```

Comment:
This is not needed I think.

Reply:
Somehow, this was necessary for the E2E test. See [this comment](https://github.com/WordPress/gutenberg/pull/48208#issuecomment-1435988885) and [this comment](https://github.com/WordPress/gutenberg/pull/48208#issuecomment-1436014572).

Reply:
I've added it back [here](https://github.com/WordPress/gutenberg/pull/48240/commits/87b1109c0517b27ffe8f5bd5fb417a37e7a822d4) since the specs failed without it.

Reply:
Any idea why it would fail in CI but not locally?

Reply:
Maybe you started the PR before my refactoring PR was opened. What if you rebase?

Reply:
This branch was created from a recent trunk (last commit is https://github.com/WordPress/gutenberg/commit/8f9c3165c5131827fdd8c73a6c1a51d29c0aeb55), so I don't think it would be the case.

Reply:
The tests have passed with this addition. I'll rerun 2-3 more times to be sure it's not a fluke.

Reply:
@youknowriad, it passed 3 times in a row with this exact addition. I'll remove the path param and see if it still passes.

Reply:
It failed. Re-adding back to be extra sure it fixes the issue.

Reply:
Weird, I'm fine keeping it for now but we'll have to figure out the reason because thet route don't exist anymore. Maybe the presence of the "path" argument triggers some condition.

Reply:
I am not sure I understand it in depth but is this line relevant?
https://github.com/WordPress/gutenberg/blob/aa1fca2c794a78b5153de07c5d6c8863f5c81795/packages/edit-site/src/components/sync-state-with-url/use-sync-path-with-url.js#L67

Reply:
I found a thing of interest. In **Firefox**, accessing the following URL will cause a redirect and the page will not be displayed correctly.

From:
```
http://localhost:8888/wp-admin/site-editor.php?canvas=view&postType=page&postId=2
```

Redirected to:
```
http://localhost:8888/wp-admin/site-editor.php?postId=none
```

![firefox](https://user-images.githubusercontent.com/54422211/220130147-435d5f4c-cf60-4214-ac02-64cfe60a8e84.png)

Reply:
Good find. 

So we do have [this code](https://github.com/WordPress/gutenberg/blob/aa1fca2c794a78b5153de07c5d6c8863f5c81795/lib/compat/wordpress-6.2/site-editor.php#L34-L45 ) in the Gutenberg plugin that forces the site editor to redirect to "postId:none". But the reasoning for that code is that it's a filter for `wp_redirect` meaning it only runs if we're redirecting to the site editor. Something we do in WP core (in 6.1 and before) but that redirect is not supposed to happen if we provide a postId and postType in the URL https://github.com/WordPress/wordpress-develop/blob/73f7aa4a149890f2e565ac74d9d49b5258b8b984/src/wp-admin/site-editor.php#L36-L46

So the question here is why the redirect filter is triggering while we do have postId and postType in the URL.

Reply:
Tests passed! 🥳 I need to head out now, so feel free to ship it.

Reply:
Okay, I found out why this is happening on CI. I think it's because it doesn't have some required cookies, yet. You can reproduce it locally in incognito mode. Hope that it will spark some ideas for the solution.

Meanwhile, I think we should just merge this to unblock other PRs. We can figure out why and the proper solution in another PR. WDYT?

Reply:
How about trying to remove the code in [this comment](https://github.com/WordPress/gutenberg/pull/48240#discussion_r1111824772) before merging?

Reply:
@youknowriad 

> I found a thing of interest. In Firefox, accessing the following URL will cause a redirect and the page will not be displayed correctly.

Sorry, this was my mistake. I was simply not logged into WordPress 😅 Therefore, this problem also occurs in Chrome. This issue may need to be looked into separately, as this URL should originally redirect to the login page.